File: JavaScriptBasics/Problem3.js
/* 
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
*/


var checkIfPrime = function(num){

	for (var i = 2; i < num; i++){
		if(num % i === 0)
			return false;
	}
	else {
		return true;
	}
};


var largestPrime = function (num) {

	var divisor = num - 1;

	while((divisor) > 1) {
		if (num % (divisor) === 0 && checkIfPrime(divisor)) {
			return divisor;
		}
		divisor--;
	}
	return num;

};

lagestPrime(2134234);

File: ruby_basics/problem3.rb

def check_if_prime (number)
	(2...number).step(1) do |i|
		if number % i == 0
			return false
		else
			return true
		end
	end
end

def largest_prime (number)
	divisor = number - 1

	while divisor > 1 do
		if ((number % divisor == 0) && check_if_prime(divisor))
			return divisor
		end
		divisor = divisor - 1
	end
	return number
end

puts largest_prime(1123123)